wgpu-hal/src/dx12/mod.rs

```diff
@@ -1092,38 +1091,33 @@ impl crate::Surface for Surface {
 
                 let swap_chain1 = swap_chain1.map_err(|err| {
                     log::error!("SwapChain creation error: {}", err);
-                    crate::SurfaceError::Other("swap chain creation")
+                    crate::SurfaceError::Other("swapchain creation")
                 })?;
 
                 match &self.target {
                     SurfaceTarget::WndHandle(_) | SurfaceTarget::SurfaceHandle(_) => {}
                     SurfaceTarget::Visual(visual) => {
-                        if let Err(err) = unsafe { visual.SetContent(&swap_chain1) }.into_result() {
-                            log::error!("Unable to SetContent: {}", err);
+                        if let Err(err) = unsafe { visual.SetContent(&swap_chain1) } {
+                            log::error!("Unable to SetContent: {err}");
                             return Err(crate::SurfaceError::Other(
                                 "IDCompositionVisual::SetContent",
                             ));
                         }
                     }
                     SurfaceTarget::SwapChainPanel(swap_chain_panel) => {
-                        if let Err(err) =
-                            unsafe { swap_chain_panel.SetSwapChain(&swap_chain1) }.into_result()
-                        {
-                            log::error!("Unable to SetSwapChain: {}", err);
+                        if let Err(err) = unsafe { swap_chain_panel.SetSwapChain(&swap_chain1) } {
+                            log::error!("Unable to SetSwapChain: {err}");
                             return Err(crate::SurfaceError::Other(
                                 "ISwapChainPanelNative::SetSwapChain",
                             ));
                         }
                     }
                 }
 
-                match swap_chain1.cast::<Dxgi::IDXGISwapChain3>() {
-                    Ok(swap_chain3) => swap_chain3,
-                    Err(err) => {
-                        log::error!("Unable to cast swap chain: {}", err);
-                        return Err(crate::SurfaceError::Other("swap chain cast to 3"));
-                    }
-                }
+                swap_chain1.cast::<Dxgi::IDXGISwapChain3>().map_err(|err| {
+                    log::error!("Unable to cast swapchain: {}", err);
```

Comment:
nitpick(non-blocking): You've been so proactive in converting this code to use `fmt` shorthand, it'd be a shame if we didn't use `{err}` here. 🤪

Reply:
Too focused on breaking my brain on the broken grammar in these messages :rofl: